src/core/Mixer.cpp

```diff
@@ -292,7 +297,7 @@ bool Mixer::criticalXRuns() const
 
 void Mixer::pushInputFrames( sampleFrame * _ab, const f_cnt_t _frames )
 {
-	lockInputFrames();
+	requestChangeInModel();
```

Comment:
I have explained the [architecture](https://github.com/LMMS/lmms/wiki/LMMS-Architecture#user-content-synchronization-with-the-mixer) in the Wiki. In this instance, rather than doing the change as soon as possible, the change is deferred until the mixer executes `runChangesInModel()`. This way, the lock in the mixer: